jobbergate-api/jobbergate_api/apps/job_submissions/routers.py

```diff
@@ -259,15 +260,22 @@ async def job_submission_agent_update(
 
     job_state_details = slurm_job_state_details[update_params.slurm_job_state]
     if job_state_details.is_abort_status:
-        # The job is aborted and should notify the user...when we have notifications for Jobbergate
         update_dict["status"] = JobSubmissionStatus.ABORTED
         update_dict["report_message"] = update_params.slurm_job_state_reason
     elif job_state_details.is_done_status:
-        # The job is completed and should notify the user...when we have notifications for Jobbergate
         update_dict["status"] = JobSubmissionStatus.DONE
 
     job_submission = await secure_services.crud.job_submission.update(id, **update_dict)
 
+    if job_submission.status in (
+        JobSubmissionStatus.ABORTED,
+        JobSubmissionStatus.DONE,
+    ):
+        await publish_status_change(
```

Comment:
> Could it be done by a background tasks? The response could be sent quicker, there is no need to wait for the notification to be processed.

the notification itself takes milliseconds to execute, so I don't see a solid reason to make it in a background task. Also, the possible error is in case the RabbitMQ endpoint is not available, and for that, there are other methods to avoid (such as define dependencies on docker compose or liveness probe in a Kubernetes environment)

Reply:
So, first, it doesn't need to be a background task because, like Tosta mentioned, it is very, very fast to publish a message to Rabbitmq and it is all async.

Second, all dangerous logic in the `publish_status_chanbe` method is wrapped by a try/catch that logs an error and continues. So, if the publish fails for some reason, the response will not be impacted.

Reply:
Fair enough